### PR TITLE
Design of Range

### PR DESCRIPTION
**Pull request by LilianaChevka**

### CODE REVIEW

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 1]
```diff
@@ -1,60 +1,126 @@
...
+    public static class FloatNextValue implements Function<Float, Float> {
+        @Override
+        public Float apply(Float f) {
+            return f + 0.1f;
+        }
+    }
+
+    public static Range<Float> of(Float first, Float last) {
+        Function<Float, Float> fnv = new FloatNextValue();
+        Range<Float> floatRange = Range.of(first, last, fnv);
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: You don't use this variable, return value immediately

File: src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package co.inventorsoft.academy.collections.model;
+     package co.inventorsoft.academy.collections.model;
```
Comment (george-stotle) [inline src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: Format code

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 3]
```diff
@@ -1,60 +1,126 @@
...
+        return floatRange;
+    }
+
+    public static <T extends Comparable<T>> Range of(T first, T last, Function<T, T> func) {
+        Range<T> tRange = new Range<>();
+        if (first.compareTo(last) < 0) {
+            T tVal = first;
+            while (tVal.compareTo(last) < 1) {
+                tRange.add(tVal);
+                System.out.println(tVal);
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: Remove souts

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 4]
```diff
@@ -1,60 +1,126 @@
...
+        @Override
+        public Integer apply(Integer i) {
+            return i + 1;
+        }
+    }
+
+
+    public static Range<Integer> of(Integer first, Integer last) {
+        IntegerNextValue inv = new IntegerNextValue();
+        Range<Integer> rangeOf = Range.of(first, last, inv);
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: You don't use rangeOf variable, return value immediately

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 5]
```diff
@@ -1,60 +1,126 @@
...
+    public final List<T> expectedElements = new ArrayList<>();
+
+    public static class IntegerNextValue implements Function<Integer, Integer> {
+
+        @Override
+        public Integer apply(Integer i) {
+            return i + 1;
+        }
+    }
+
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:28]: Format code

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 6]
```diff
@@ -1,60 +1,126 @@
...
+        return expectedElements.retainAll(c);
     }
 
     public boolean removeAll(Collection<?> c) {
-        return false;
+        return expectedElements.removeAll(c);
     }
 
     public void clear() {
+        expectedElements.clear();
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: remove line

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 7]
```diff
@@ -1,60 +1,126 @@
...
 public class Range<T> implements Set<T> {
 
+    @Override
+    public String toString() {
+        return "Range{" +
+                "expectedElements=" + expectedElements +
+                '}';
+    }
+
+    public final List<T> expectedElements = new ArrayList<>();
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: make it private

**Review by george-stotle — CHANGES_REQUESTED** [review]

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 1 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: ok

File: src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java (hunk 2 above)
Comment (LilianaChevka) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: ok

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 3 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: ok

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 4 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: ok

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 5 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:28]: ok

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 6 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: ok

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 7 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: ok

File: src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java (hunk 2 above)
Comment (george-stotle) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: Does ok mean that is done or not?

Comment (LilianaChevka) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: Yes, It's done

Comment (george-stotle) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: Format code, it's not done.

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 8]
```diff
@@ -1,60 +1,119 @@
...
+    private final List<T> expectedElements = new ArrayList<>();
+
+    public static class IntegerNextValue implements Function<Integer, Integer> {
+
+        @Override
+        public Integer apply(Integer i) {
+            return i + 1;
+        }
+    }
+
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:28]: Remove redundant line

**Review by george-stotle — CHANGES_REQUESTED** [review]

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 8 above)
Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:28]: it's done

File: src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java (hunk 2 above)
Comment (LilianaChevka) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: Now it's done.

Comment (george-stotle) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: It's not

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 9]
```diff
@@ -1,60 +1,117 @@
...
-        return false;
+        return expectedElements.removeAll(c);
     }
 
     public void clear() {
-
+        expectedElements.clear();
     }
-}
+}
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: Add new line at the end of file

Comment (LilianaChevka) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: ok

**Review by george-stotle — APPROVED** [review]

File: src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java (hunk 2 above)
Comment (LilianaChevka) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: it's ok now?

Comment (george-stotle) [reply src/main/test/co/inventorsoft/academy/collections/model/RangeTest.java]: yep

**Pull request closed without merging**